corehq/apps/users/views/__init__.py

```diff
@@ -549,6 +553,7 @@ def page_context(self):
 
 @method_decorator(always_allow_project_access, name='dispatch')
 @method_decorator(require_can_edit_or_view_web_users, name='dispatch')
+@location_safe
```

Comment:
Looking at this page, I see it also includes links for:
* Invite Web User
* Download Web Users
* Upload Web Users
* Remove Membership
* Resend Invitation
* Delete Invitation

(possibly more, depending on what's enabled for the project, I'm not sure that's comprehensive)
I think each of those will either need to be also made location-safe or hidden from view.  As-is, I'd expect a restricted user to get a 403 if they click any of those buttons.  Not sure which functions we plan on implementing at this time, though (I think bulk actions were called out of scope?)